replay2/memory_remapper/tests/memory_remapper_tests.cc

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2022 Google Inc.
+//
+// This file is generated code. It was created by the AGI code generator.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include "replay2/memory_remapper/memory_remapper.h"
+
+using namespace agi::replay2;
+
+class ModResourceGenerator : public ResourceGenerator {
+public:
+	ModResourceGenerator(size_t length) : length_(length) {}
+	virtual ~ModResourceGenerator() {}
+
+	size_t length() const override { return length_; }
+
+	void generate(ReplayAddress replayAddress) override {
+		for(size_t i = 0; i < length_; ++i) {
```

Comment:
Shouldn't we generate this in the test itself because to be able to understand the test, someone has to jump here and see the values.

Reply:
Think of them as test helpers. It's common to have test helpers so reduce repetition in code.

Reply:
I see. I was just hoping that we could give the fill value as a parameter to generate, so it would be clear to what we are actually testing against.

Reply:
That's basically what we're doing here. The type and the constructor args dictate what we're filling with. generate() itself doesn't take any args because it's a general interface for the real program. In future we'll have ResourceGenerators that read data out of gfxtrace files and other things.